Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -435,7 +435,10 @@ public class PBXProjGenerator {
 
                 dependencies.append(targetDependency.reference)
 
-                if (dependencyTarget.type.isLibrary || dependencyTarget.type.isFramework) && dependency.link {
+                let dependecyLinkage = dependencyTarget.defaultLinkage
+                let shouldLink = (dependecyLinkage == .dynamic && target.type != .staticLibrary)
+                    || (dependecyLinkage == .static && target.type.isExecutable)
+                if dependency.link ?? shouldLink {
```

Comment:
Once you're checking for build settings, then that code shouldn't run if they have manually specified link setting, to aid in performance

Reply:
Good point. I’ll adjust.